libqtile/core/manager.py

```diff
@@ -530,19 +539,52 @@ def current_window(self):
     def add_strut(self, strut):
         from libqtile.bar import Bar, Gap
 
+        try:
+            if strut[0]:
+                x = strut[0]
+            elif strut[1]:
+                x = self.x_screen_dimensions.get('x') - strut[1]
+            else:
+                x = ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+            if strut[2]:
+                y = strut[2]
+            elif strut[3]:
+                y = self.x_screen_dimensions.get('y') - strut[2]
+            else:
+                y = ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
```

Comment:
I am not too fond of this bulky statement, would you consider something like
```suggestion
            x = strut[0] or \
                (strut[8] + strut[9]) / 2 or \
                (strut[10] + strut[11]) / 2 or \
                self.x_screen_dimensions.get('x') - strut[1]
            y = strut[2] or \
                (strut[4] + strut[5]) / 2 or \
                (strut[6] + strut[7]) / 2 or \
                self.x_screen_dimensions.get('y') - strut[2]
```

to be more readable or do all of those or connections make things worse?

Reply:
You suggested to switch the order of the `elif` and the `else` block. This changes the behaviour! I don't know the spec, but this exchange in the execution order looks suspicious, so I thought it'd be better to point it out. However, I have no clue which of the two (different!) implementations is in agreement with the spec. If it is the original bulky block, an alternative would be
```py
x = (strut[0] if strut[0] else
     (self.x_screen_dimensions.get('x') - strut[1] if strut[1] else
      ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)))
y = (strut[2] if strut[2] else
     (self.x_screen_dimensions.get('y') - strut[3] if strut[3] else
      ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)))
```
or 
```py
x = (strut[0] if strut[0]
     else (self.x_screen_dimensions.get('x') - strut[1] if strut[1]
           else ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)))
y = (strut[2] if strut[2]
     else (self.x_screen_dimensions.get('y') - strut[3] if strut[3]
           else ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)))

```
with the indentation shifted to the right of course.
Also, in line 545-546 you use index `1` two times, and in 551-552 you use `3` and `2` where the rest of the lines is similar. Is that correct?

Reply:
As far as I can see the order should not matter as any of the values, if set to something other than zero, give an idea of on-screen coordinates. A problem would be one window setting struts on different screens but if that happens neither implementation is correct.
Good catch with the indices, I agree line 551 should be `y = self.x_screen_dimensions.get('y') - strut[2]`